Move Previous API Docs to the RKA Section (#1147)
rancher/rancher-docs: docs/api/api-tokens.md

---
title: Using API Tokens
---

<head>
  <link rel="canonical" href="https://ranchermanager.docs.rancher.com/api/previous-rancher-api-versions/api-tokens"/>
</head>

Rancher `v2.8.0` introduced the [Rancher Kubernetes API](./api-reference.mdx) which can be used to manage Rancher resources through `kubectl` or the [web interface](./v3-rancher-api-guide.md#enable-view-in-api).

By default, some cluster-level API tokens are generated with infinite time-to-live (`ttl=0`). In other words, API tokens with `ttl=0` never expire unless you invalidate them. Tokens are not invalidated by changing a password.

You can deactivate API tokens by deleting them or by deactivating the user account.

> ### Deleting tokens

Reviewer: ```suggestion
## Deleting Tokens

```

> ### Deleting tokens

Reviewer: I made some edit suggestions to sync with #1184, but in some cases sections were moved around as well as edited, which isn't so easy to suggest from the Github web interface. 

The PR touched 4 versions of this page and the v2.7 and v2.6 differ slightly from latest, so those also need to be synced.

> ### Deleting tokens

Author: Just synced the api-tokens across versions, thanks Marty!

1. Access the token you want to delete by its ID. For example, `https://<Rancher-Server-IP>/v3/tokens/kubectl-shell-user-vqkqt`

1. Click **Delete**.

Here is the complete list of tokens that are generated with `ttl=0`:

| Token             | Description                                                                            |
| ----------------- | -------------------------------------------------------------------------------------- |
| `kubeconfig-*`    | Kubeconfig token                                                                       |
| `kubectl-shell-*` | Access to `kubectl` shell in the browser                                               |
| `agent-*`         | Token for agent deployment                                                             |
| `compose-token-*` | Token for compose                                                                      |
| `helm-token-*`    | Token for Helm chart deployment                                                        |
| `telemetry-*`     | Telemetry token                                                                        |

> | `drain-node-*`    | Token for drain (we use `kubectl` for drain because there is no native Kubernetes API) |

Reviewer: ```suggestion
| `drain-node-*` | Token for drain (Rancher uses `kubectl` for drain because there is no native Kubernetes API). |
```

---
title: Using API Tokens
---

<head>
  <link rel="canonical" href="https://ranchermanager.docs.rancher.com/api/api-tokens"/>
</head>

> Rancher v2.8.0 introduced the [Rancher Kubernetes API](./api-reference.mdx) which can be used to manage Rancher resources through `kubectl` or the [web interface](./v3-rancher-api-guide.md#enable-view-in-api).

Reviewer 2: The RK API is not accessible through the web interface referenced since the latter is the v3 API.

We also need to establish the (lack of) relationship between the token content on this page with the RK API and v3 API. Maybe some variation of the below (insert links where needed)?

```suggestion
Rancher v2.8.0 introduced the [Rancher Kubernetes API](./api-reference.mdx) which can be used to manage Rancher resources through `kubectl`. This page covers information on API tokens used with the Rancher CLI, kubeconfig files, Terraform, and the [v3 API browser](./v3-rancher-api-guide.md#enable-view-in-api).
```

Author: Updated with suggestions and added in links where I thought appropriate, thanks for reviewing Billy!

## Disable Tokens in Generated Kubeconfigs

Set the `kubeconfig-generate-token` setting to `false`. This setting instructs Rancher to no longer automatically generate a token when a user clicks on download a kubeconfig file. When this setting is deactivated, a generated kubeconfig references the [Rancher CLI](../reference-guides/cli-with-rancher/kubectl-utility.md#authentication-with-kubectl-and-kubeconfig-tokens-with-ttl) to retrieve a short-lived token for the cluster. When this kubeconfig is used in a client, such as `kubectl`, the Rancher CLI needs to be installed to complete the log in request.

## Token Hashing

> You can [enable token hashing](../how-to-guides/advanced-user-guides/enable-experimental-features/enable-experimental-features.md), where tokens undergo a one-way hash using the SHA256 algorithm. This is a non-reversible process: once enabled, this feature cannot be disabled. It is advisable to take backups before enabling and/or evaluating in a test environment first.

Reviewer: ```suggestion
You can [enable token hashing](../how-to-guides/advanced-user-guides/enable-experimental-features/enable-experimental-features.md), where tokens undergo a one-way hash using the SHA256 algorithm. This is a non-reversible process: once enabled, this feature cannot be disabled. You should first evaluate this setting in a test environment, and/or take backups before enabling.
```

Reword for clarity

Author: Just updated across versions, thanks Marty!

## Disable Tokens in Generated Kubeconfigs

Set the `kubeconfig-generate-token` setting to `false`. This setting instructs Rancher to no longer automatically generate a token when a user clicks on download a kubeconfig file. When this setting is deactivated, a generated kubeconfig references the [Rancher CLI](../reference-guides/cli-with-rancher/kubectl-utility.md#authentication-with-kubectl-and-kubeconfig-tokens-with-ttl) to retrieve a short-lived token for the cluster. When this kubeconfig is used in a client, such as `kubectl`, the Rancher CLI needs to be installed to complete the log in request.

## Token Hashing

> Users can enable token hashing, where tokens undergo a one-way hash using the SHA256 algorithm. This is a non-reversible process: once enabled, this feature cannot be disabled. It is advisable to take backups prior to enabling and/or evaluating in a test environment first.

Reviewer: ```suggestion
You can [enable token hashing](../how-to-guides/advanced-user-guides/enable-experimental-features/enable-experimental-features.md), where tokens undergo a one-way hash using the SHA256 algorithm. This is a non-reversible process: once enabled, this feature cannot be disabled. It is advisable to take backups prior to enabling and/or evaluating in a test environment first.
```

Author: Updated with suggestion

> To enable token hashing, refer to [this section](../how-to-guides/advanced-user-guides/enable-experimental-features/enable-experimental-features.md).

Reviewer: ```suggestion
```

Linking to page in previous line, upon mention of enabling the feature

Author: Updated and removed
